test/powershell/engine/Api/BasicEngine.Tests.ps1

```diff
@@ -21,3 +21,29 @@ Describe 'Basic engine APIs' -Tags "CI" {
         }
     }
 }
+
+Describe "Clean up open Runspaces when exit powershell process" -Tags "Feature" {
+    It "PowerShell process should not freeze at exit" {
+        $command = @'
+-c $rs = [runspacefactory]::CreateRunspacePool(1,5)
+$rs.Open()
+$ps = [powershell]::Create()
+$ps.RunspacePool = $rs
+$null = $ps.AddScript(1).Invoke()
+write-host should_not_hang_at_exit
+exit
+'@
+        $process = Start-Process pwsh -ArgumentList $command -PassThru
+        for ($i = 0; -not $process.HasExited -and $i -lt 5; $i++) {
```

Comment:
Please use the helper at: https://github.com/PowerShell/PowerShell/blob/master/test/tools/Modules/HelpersCommon/HelpersCommon.psm1#L1

Reply:
OK, I will use `Wait-UntilTrue`